File: ChatDemo-UI3.0/ChatDemo-UI3.0/Code/Controllers/Settings/EMChatsSettingViewController.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */

#import "EMChatsSettingViewController.h"

@interface EMChatsSettingViewController ()
@property (nonatomic, strong) UISwitch *autoAcceptSitch;
@property (nonatomic) BOOL autoAcceptInvitation;
@end

@implementation EMChatsSettingViewController

- (UISwitch *)autoAcceptSitch
{
    if (!_autoAcceptSitch) {
        _autoAcceptSitch = [[UISwitch alloc] init];
        [_autoAcceptSitch addTarget:self action:@selector(autoAcceptAction:) forControlEvents:UIControlEventValueChanged];
    }
    return _autoAcceptSitch;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self configBackButton];
    [self loadOptions];
    
}

- (void)loadOptions
{
    EMOptions *options = [[EMClient sharedClient] options];
    _autoAcceptInvitation = options.isAutoAcceptGroupInvitation;
    [self.autoAcceptSitch setOn:_autoAcceptInvitation animated:YES];
    [self.tableView reloadData];
}



- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 1;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *cellIdentifier = @"ChatsCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    if (!cell) {
        
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:cellIdentifier];
    }
    cell.textLabel.text = NSLocalizedString(@"setting.chats.acceptInvitation", @"Accept group invites automatically");
    self.autoAcceptSitch.frame = CGRectMake(tableView.frame.size.width - 65, 8, 50, 30);
    [cell.contentView addSubview:self.autoAcceptSitch];

    return cell;
}

- (void)autoAcceptAction:(UISwitch *)sender
{
    EMOptions *options = [[EMClient sharedClient] options];
    options.isAutoAcceptGroupInvitation = sender.isOn;
}

@end

File: swift-demo/Hyphenate-Demo-Swift/Code/Controllers/Settings/EMChatsSettingViewController.swift
//
//  EMChatsSettingViewController.swift
//  Hyphenate-Demo-Swift
//
//  Created by 杜洁鹏 on 2017/7/7.
//  Copyright © 2017年 杜洁鹏. All rights reserved.
//

import UIKit
import Hyphenate

class EMChatsSettingViewController: EMBaseSettingController {
    
    var autoAcceptInvitation = false
    lazy var autoAcceptSwitch = {()-> UISwitch in
        let switchCotrol = UISwitch()
        switchCotrol.addTarget(self, action: #selector(autoAcceptAction), for: UIControlEvents.valueChanged)
        return switchCotrol
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        configBackButton()
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCell(withIdentifier: "AcceptInviteCell")
        if cell == nil {
            cell = UITableViewCell.init(style: UITableViewCellStyle.value1, reuseIdentifier: "AcceptInviteCell")
        }
        
        cell?.textLabel?.text = "Accept group invites automatically"
        autoAcceptSwitch.frame = CGRect.init(x: tableView.width() - 65, y: 8, width: 50, height: 30)
        cell?.contentView.addSubview(autoAcceptSwitch)
        
        return cell!
    }
    
    func loadOptions() {
        let options = EMClient.shared().options
        autoAcceptInvitation = (options?.isAutoAcceptGroupInvitation)!
        autoAcceptSwitch.setOn(autoAcceptInvitation, animated: true)
        tableView.reloadData()
    }
    
    func autoAcceptAction(_ sender: UISwitch) {
        let options = EMClient.shared().options
        options?.isAutoAcceptGroupInvitation = sender.isOn
    }
}
